### PR TITLE
feat: pytest

### PR DESCRIPTION
**Pull request by idowuseyi**
**What does the PR do?**

This PR streams the API endpoint compatibility test.

**How should this be manually tested?**

1. Send a GET request to `/run-tests`

Successful Response

On a successful response, the scripts is run and displayed as the response to the user.

 **Checklist of what has been done**
- [x] Define API endpoint to run-tests.
- [x]  Run a python test scripts to run API endpoints.
- [x]  Stream the responses.

### CODE REVIEW

**Review by Heba-WebDev — APPROVED** [review]

File: src/tests/tests.controller.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { Controller, Get, Res } from '@nestjs/common';
+import { Response } from 'express';
+import { TestsService } from './tests.service';
+
+@Controller('tests')
```
Comment (CFCIfe) [inline src/tests/tests.controller.ts:5]:
```suggestion
@Controller('run-tests')
```

File: src/tests/tests.controller.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+export class TestsController {
+  constructor(private readonly testsService: TestsService) {}
+
+  @Get('run-tests')
```
Comment (CFCIfe) [inline src/tests/tests.controller.ts:9]:
```suggestion
@Get()
```

File: src/tests/tests.controller.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  runTests(@Res() res: Response) {
+    const stream = this.testsService.runTests();
+    stream.pipe(res);
```
Comment (Cobidex) [inline src/tests/tests.controller.ts:12]: better to use pipeline, for better management of buffer and graceful end of the stream

File: src/tests/tests.service.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { Injectable } from '@nestjs/common';
+import { spawn } from 'child_process';
+import { Readable } from 'stream';
+
+@Injectable()
+export class TestsService {
+  runTests(): Readable {
+    const process = spawn('python', ['path/to/your/script.py']);
```
Comment (Cobidex) [inline src/tests/tests.service.ts:8]: Remove the dummy path.
Provide the path to your script through an environment variable. Or the process.cwd() API.

Comment (idowuseyi) [reply src/tests/tests.service.ts:8]: Thanks

File: src/tests/tests.controller.ts (hunk 3 above)
Comment (idowuseyi) [reply src/tests/tests.controller.ts:12]: Thanks.

File: src/tests/tests.controller.ts (hunk 2 above)
Comment (idowuseyi) [reply src/tests/tests.controller.ts:9]: Good

File: src/tests/tests.controller.ts (hunk 1 above)
Comment (idowuseyi) [reply src/tests/tests.controller.ts:5]: Thanks

**Pull request closed without merging**